pkg/ebpf/events_pipeline.go

```diff
@@ -282,15 +282,21 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 }
 
 // matchPolicies does the userland filtering (policy matching) for events. It iterates through all
-// existing policies, that were set by the kernel in the event bitmask. Some of those policies might
+// existing policies, that were set by the kernel in the event bitmap. Some of those policies might
 // not match the event after userland filters are applied. In those cases, the policy bit is cleared
 // (so the event is "filtered" for that policy).
 // This may be called in different stages of the pipeline (decode, derive, engine).
 func (t *Tracee) matchPolicies(event *trace.Event) uint64 {
 	eventID := events.ID(event.EventID)
 	bitmap := event.MatchedPoliciesKernel
 
-	for p := range t.config.Policies.Map() { // range through each existing policy
+	// Short circuit if there are no policies in userland that need filtering.
+	if bitmap&t.config.Policies.FilterableInUserland() == 0 {
+		event.MatchedPoliciesUser = bitmap // store untoched bitmap to be used in sink stage
+		return bitmap
+	}
```

Comment:
You can place this logic in FilterableInUserlandMap() and have it iterate 0 items (instead of doing extra logic outside the object).